### ISSUE TITLE
Add native support for ActiveRecord Enums

### ISSUE THREAD

**Issue by AlexeyMK** [activeadmin/activeadmin#3157]
Rails 4.1 now [has support for enums](http://api.rubyonrails.org/v4.1.0/classes/ActiveRecord/Enum.html).

It seems pretty straightforward to default enum columns to a select in the edit form window: http://stackoverflow.com/questions/23414880/how-to-properly-configure-rails-4-1-enums-in-activeadmin/23799562#23799562

Would be pretty awesome to support this natively, without requiring additional customization.

**Comment by stefanvignir** [comment]
@justinfrench has implemented this already according to https://github.com/justinfrench/formtastic/pull/1098. Any news about the current status as of now or a workaround that works for edit forms?

**Comment by timoschilling** [comment]
Here are no news. Maybe some want to work on this?

**Comment by thom-nic** [comment]
For the record for other folks coming across this issue, the workaround is to explicitly specify the input for that column as a `select` with the correct values like so:

```ruby
  form do |f|
    f.semantic_errors *f.object.errors.keys
    f.inputs do
      f.input :status, as: :select, collection: MyModel.statuses.keys
      # other attribute fields...
    end
    f.actions
  end
```

...where `status` is the enum column and `statuses` is a hash that ActiveRecord automatically generates on the model from the enum values.

**Comment by XBeg9** [comment]
@thom-nic Do you know how to filter use enums in filter? Thanks

**Comment by besi** [comment]
:+1: 

@XBeg9: I found this here <http://stackoverflow.com/a/25437896/784318>:

    filter :level, as: :select, collection: Model.levels

**Comment by iangreenleaf** [comment]
>     filter :level, as: :select, collection: Model.levels

This solution makes the filtering functionality work, but the "Current filters" section displays the integer values, like "Status equals: 2". And unfortunately using the `keys` as suggested for the other solutions won't work here, because ActiveRecord::Enum doesn't support `Model.where(status: "foobar")`.

So, one more small thing that it would be great to have AA do automatically if possible.

**Comment by gerryhd** [comment]
> This solution makes the filtering functionality work, but the "Current filters" section displays the integer values, like "Current filters: Status equals 2"

I somewhat looked into a way to implement this. It seems that a method `defined_enums` exists. I'm thinking maybe this can be evaluated in the values method in active_filter.rb to then retrieve the values if the key is found in the defined_enums hash.

**Comment by sergey-alekseev** [comment]
Just FYI this is still actual in 2021.

**Comment by sebfie** [comment]
Right, the issue is still here : 
![CleanShot 2022-01-20 at 16 29 46](https://user-images.githubusercontent.com/862129/150369433-b4707fea-1eb2-41f7-9f24-ffe3040b3c5d.png)

**Comment by harunkumars** [comment]
> I somewhat looked into a way to implement enum keys into the filter message. It seems that a method `defined_enums` exists for models. I'm thinking maybe this can be evaluated in the values method in active_filter.rb to then retrieve the values if the key is found in the defined_enums hash.

This PR (https://github.com/activeadmin/activeadmin/pull/7790) codifies the comment above from @gerryhd

**Comment by javierjulio** [comment]
We have a PR for this. Thank you.

**Comment by npgretz** [comment]
This is still very much a desired feature for me. I inherited an application where a combination of Enumerize and enum are used with ActiveAdmin filter and input forms.

For ActiveRecord models with:

`enum status: { 'completed': 0, 'pending': 1}`

I cannot use:

`filter :status, as: :select, multiple: true ...`

and the int is shown in the view of active filters. Alternatively, some of the models have varchar (int was used for models with enum) for their status column type in MySQL and use:

`enumerize :status, in: {to_be_scheduled:0, scheduled:1, completed:2, incomplete:4, left_message:5, unreachable:6}, scope: :statuses`

For these models, I can use:

`filter :status, as: :select, multiple: true ...`

and in the view of active filters I will see "Status in scheduled", instead of "Status in 1" that the models using enum show. This was all very confusing to me when I inherited this application and I did not find any help in documentation or online.This may be a helpful work around until the PR is merged.
